### ISSUE TITLE
Possible issue with pass-trough props from React JS component

### ISSUE THREAD

**Issue by TomiS** [cristianoc/genType#126]
I'm trying to use genType to manage types in a following example application:
```js
<App> // Written in Reason
  <ReactJsComponent> // Written in Js/Flow
    <Button icon='danger' /> // Written in Reason, renders optional Icon as a child if icon prop is set
    <Icon iconType='danger' /> // Written in Reason
  </ReactJsComponent>
</App>
```

It seems Icon renders a correct svg shape only when called directly from ReactJsComponent. If I try to create a Button with icon, it renders wrong icon svg shape because 'danger' is passed trough as string and is not converted to integer mapping.

Everything works when called from Reason, so I'm tempted to suspect there is something wrong with how genType works in this situation.

A quite minimal repo reproducing the problem can be found here:
https://github.com/TomiS/reasontest

How to install and run (requires yarn)
```bash
git clone https://github.com/TomiS/reasontest.git
cd reasontest
yarn install
yarn run clean
yarn run start
yarn run webpack
# Open browser at http://localhost:9000/
# Both icons should be same. The one inside the button is incorrect.
```

**Comment by cristianoc** [comment]
Thanks for the report! It took a couple of attempts to reproduce.

1) Design choice:
```reason
[@genType]
type t = int
and tt = string;
```
the single annotation is intended to cover both cases.

2) If in another file
```reason
type t = int;
[@genType]
type tt = string;
```
when the type declarations are imported from that file, the AST actually looks like this:
```reason
type t = int;
[@genType]
and tt = string;
```
Then the design choice from 1) kicks in, the annotation is ignored, and both `t` and `tt` are considered to be without annotations.

In this diff, https://github.com/cristianoc/genType/commit/29ce1e64a65a8afea2da3899fd177028bfa9e2bb, the design choice 1) is modified so that each entry in a list of type definitions takes its own annoation.

**Comment by cristianoc** [comment]
@TomiS Let me know if you're able to try master first, or I'll just create a bugfix release.

**Comment by TomiS** [comment]
@cristianoc Hmm.. I'm trying to test it. I installed master but now it doesn't generate any files. Instead I get a dump like this:
```
tomis@tomis-computer:~/Documents/Git/reasontest$ yarn run build
yarn run v1.10.1
$ export BS_CMT_POST_PROCESS_CMD="gentype" && bsb -make-world
[18/18] Building src/ReactDOMRe.mlast.d
[1/9] Building src/ReasonReact.cmi
sh: 1: gentype: not found
[2/9] Building src/ReasonReactOptimizedCreateClass.cmj
sh: 1: gentype: not found
[3/9] Building src/ReactEvent.cmi
sh: 1: gentype: not found
[4/9] Building src/ReactEvent.cmj
sh: 1: gentype: not found
[5/9] Building src/ReactEventRe.cmi
sh: 1: gentype: not found
```

**Comment by cristianoc** [comment]
@TomiS fair enough we haven't documented much how to use a version from master by modifying the build/start commands
```json
"scripts": {
    "build": "export BS_CMT_POST_PROCESS_CMD=\"gentype\" && bsb -make-world",
```
to use the complete path to a specific gentype executable.
I'm creating a patch release 2.3.1.

**Comment by TomiS** [comment]
Tested 2.3.1. Unfortunately it does not remove my issue. I pushed latest changes to my repo.

**Comment by cristianoc** [comment]
Can you try with `"namespace": false,` in the configs.
Namespace support somehow got out of sync, as there are no example projects using it in CI.
Thart's the same reason for the type errors if you run flow.

**Comment by cristianoc** [comment]
I'll take a look at bringing namespace support back up to speed.

**Comment by TomiS** [comment]
Cool. It indeed solved the issue. Thanks a lot for the fix and help!

**Comment by TomiS** [comment]
@cristianoc I hate to say, but I don't think the namespace work quite right yet. At least if I install 2.4.0 and bring back namespace:true, my example falls back to previous problem.

**Comment by cristianoc** [comment]
@TomiS indeed there was a piece missing https://github.com/cristianoc/genType/commit/c3a22bfa3addb76e6c63897dc062ce6f4dadfebc.

**Comment by TomiS** [comment]
@cristianoc Nice. I tested master quickly and it seemed to have fixed my case.

**Comment by TomiS** [comment]
@cristianoc Oh, and a tagged release would be nice when/if you have time. (I wouldn't like to push my full filesystem path to gentype to a public git repo which seems to be necessary when using master branch)

**Comment by cristianoc** [comment]
@TomiS published as version 2.4.1.
